lxd/daemon.go

```diff
@@ -1109,6 +1109,13 @@ func (d *Daemon) init() error {
 		DebugAddress:         debugAddress,
 		MetricsAddress:       metricsAddress,
 		MetricsServer:        metricsServer(d),
+		VsockServer:          vSockServer(d),
+		VsockSupport:         true,
```

Comment:
One last thing :) (sorry)

We only need to enable the VsockSupport if instancetype.VM is supported.
So rather than enable it by default and then disable it, why not flip it on its head and detect it like this:

```
err, found := d.State().InstanceTypes[instancetype.VM] == nil
if found && err == nil {
    config.VsockSupport = true
```

That way we are utilising the default false value, and we are keeping the enabling logic all in one place.
We are also detecting the explicit presence of a nil entry, rather than relying on the default value for missing entries in a map.